pkgs/games/fteqw/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, stdenv
+, fetchsvn
+, gzip
+, libvorbis
+, libmad
+, SDL2
+, SDL2_mixer
+, libpng
+, alsa-lib
+, gnutls
+, zlib
+, libjpeg
+, vulkan-loader
+, vulkan-headers
+, speex
+, libopus
+, xorg
+, libGL
+}@attrs:
+{
+  fteqw = import ./generic.nix (rec {
+    pname = "fteqw";
+    releaseFile = pname;
+
+    buildFlags = [ "m-rel" ];
+
+    nativeBuildInputs = [
+      vulkan-headers
+    ];
+
+    buildInputs = [
+      gzip libvorbis libmad SDL2 SDL2_mixer
+      libpng alsa-lib gnutls libjpeg vulkan-loader
+      speex libopus xorg.libXrandr xorg.libXcursor
+    ];
+
+    postFixup = ''
+      patchelf $out/bin/${pname} \
+        --add-needed ${SDL2}/lib/libSDL2.so \
+        --add-needed ${libGL}/lib/libGLX.so \
+        --add-needed ${libGL}/lib/libGL.so \
+        --add-needed ${gnutls.out}/lib/libgnutls.so \
+        --add-needed ${vulkan-loader}/lib/libvulkan.so
+    '';
+
+    description = "A hybrid and versatile game engine";
+  } // attrs);
+  fteqw-dedicated = import ./generic.nix (rec {
+    pname = "fteqw-dedicated";
+    releaseFile = "fteqw-sv";
+
+    buildFlags = [ "sv-rel" ];
+
+    buildInputs = [
+      gnutls zlib
+    ];
+
+    postFixup = ''
+      patchelf $out/bin/${pname} \
+        --add-needed ${gnutls}/lib/libgnutls.so \
+    '';
+
+    description = "Dedicated server for FTEQW";
+  } // attrs);
+  fteqcc = import ./generic.nix (rec {
+    pname = "fteqcc";
+    releaseFile = pname;
+
+    buildFlags = [ "qcc-rel" ];
+
+    buildInputs = [
+      zlib
+    ];
+
+    description = "User friendly QuakeC compiler";
+  } // attrs);
```

Comment:
I have the feeling that this will break cross.

Reply:
@SuperSandro2000 I've tried building it with `nix build` for `aarch64-linux` and it succeeded. I can't test on anything but `x86_64-linux`, but it does build there at least. I'm unable to run it, but I also don't know the requirements for running an x11 OpenGL application from a foreign architecture.

Reply:
(clarification: i meant building with `nix build #packages.aarch64-linux.fteqw`, not building _on_ `aarch64-linux`)

Reply:
It does indeed not work, but the failure is:

```
-> % nix build .#pkgsCross.aarch64-multiplatform.fteqw
error: builder for '/nix/store/b0x2bsy8b3xp9zayv2gfw6si9vrr1cfq-timidity-aarch64-unknown-linux-gnu-2.15.0.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/vawcrrz8cjrh1xgi4xdycar5i9wzy4f3-aarch64-unknown-linux-gnu-binutils-2.40/bin/aarch64-unknown-linux-gnu-ld: jack_a.c:(.text+0x890): undefined reference to `jack_get_ports'
       > /nix/store/vawcrrz8cjrh1xgi4xdycar5i9wzy4f3-aarch64-unknown-linux-gnu-binutils-2.40/bin/aarch64-unknown-linux-gnu-ld: jack_a.c:(.text+0x8b8): undefined reference to `jack_port_name'
       > /nix/store/vawcrrz8cjrh1xgi4xdycar5i9wzy4f3-aarch64-unknown-linux-gnu-binutils-2.40/bin/aarch64-unknown-linux-gnu-ld: jack_a.c:(.text+0x8c8): undefined reference to `jack_connect'
       > /nix/store/vawcrrz8cjrh1xgi4xdycar5i9wzy4f3-aarch64-unknown-linux-gnu-binutils-2.40/bin/aarch64-unknown-linux-gnu-ld: jack_a.c:(.text+0x940): undefined reference to `jack_client_close'
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [Makefile:1583: timidity] Error 1
       > make[2]: Leaving directory '/build/TiMidity++-2.15.0/timidity'
       > make[1]: *** [Makefile:386: all-recursive] Error 1
       > make[1]: Leaving directory '/build/TiMidity++-2.15.0'
       > make: *** [Makefile:325: all] Error 2
       For full logs, run 'nix log /nix/store/b0x2bsy8b3xp9zayv2gfw6si9vrr1cfq-timidity-aarch64-unknown-linux-gnu-2.15.0.drv'.
error: 1 dependencies of derivation '/nix/store/a05sj6041pfj3dmnd29qqa9qfjn016wx-SDL2_mixer-aarch64-unknown-linux-gnu-2.6.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/466kp4l5ik7v1wigz5v5c6jf01cadrg0-fteqw-aarch64-unknown-linux-gnu-unstable-2022-08-09.drv' failed to build
```

Not quite what I expected, nor am I experienced enough to know how I would even begin to resolve this issue..